### PR TITLE
CVE-2013-4444

### PR DESCRIPTION
**Pull request by aaa1008**
Completed CVE-2013-4444 for Tomcat

### CODE REVIEW

**Comment by aushe02** [comment]
You could add a note for "distrustful_input" to the lesson learned.   Developers need to really carefully check things like file uploads.  Also, "complex_inputs" could apply because inputs may not match assumptions about them could also apply.

File: cves/CVE-2013-4444.yml [hunk 1]
```diff
@@ -43,15 +43,23 @@ description_instructions: |
...
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+  This vulnerability allows attackers to upload and execute JavaScript programs
+  on a Tomcat server. The Tomcat security team views this vulnerability as
+  important rather than critical because the circumstances required for it to
+  occur are limited.
+
+  The implementation allowed unrestricted reading of files without obvious
```
Comment (AdamVD) [inline cves/CVE-2013-4444.yml]: Was it not obvious that files were checked for scripting? Or were there checks missing to catch even obvious scripting exploits? This sentence is a little confusing.

File: cves/CVE-2013-4444.yml [hunk 2]
```diff
@@ -202,4 +219,12 @@ mistakes:
...
     engineering industry would find interesting.
-  answer:
+  answer: |
+    Even after doing the research, it is not very clear to me what happened that
+    lead to this vulnerability other than not checking for edge cases that
+    might cause vulnerabilities such as outdated versions of languages and
+    frameworks that might not be compatible with software we're building.
+
+    In addition, it's interesting to me that the fix wasn't focused on fixing
+    the issue but instead it was an overall "code cleanup" that included the fix
```
Comment (AdamVD) [inline cves/CVE-2013-4444.yml]: This is indeed quite interesting! Was the removed code completely un-used during "normal" code execution? It is interesting that this created a vulnerability, and provides additional justification to remove any "dead" code.

**Review by AdamVD — CHANGES_REQUESTED** [review]

**Comment by AdamVD** [comment]
+3 from me!

File: cves/CVE-2013-4444.yml (hunk 1 above)
Comment (aaa1008) [reply cves/CVE-2013-4444.yml]: reworded. Thanks!

File: cves/CVE-2013-4444.yml (hunk 2 above)
Comment (aaa1008) [reply cves/CVE-2013-4444.yml]: It was used. However, commits including the fix didn't indicate that they were for fixing the issue which I personally found odd.

**Pull request merged**